Library/Homebrew/sandbox.rb

```diff
@@ -165,6 +162,9 @@ class SandboxProfile
           (regex #"^/dev/ttys?[0-9]*$")
           )
       (deny file-write*) ; deny non-whitelist file write operations
+      (deny file-read*
+          (regex #"^/Users/[a-zA-Z0-9_]+/.*_history$")
+          )
```

Comment:
Hi, here are some advice from me:
* Using `[a-zA-Z0-9_]+` to match user name is bad. Because in fact, user name can have many other arbitrary characters, e.g. Unicode chars in other language. It is recommend to use `/[^/]+/` to match a directory.
* Using `.*_history$` is way too general. It can even match path like `~/Homebrew/Cellar/foo/1.0/.../some_script_end_with_history`. You get the idea. It would be better to specific the common shell history files verbosely without using `.*` regex.
* I would not recommend to add this rule directly in `SandboxProfile` base template. Instead, can you create `Sandbox#deny_read_shell_history` method? This can also be helpful, say not enforcing such rule under `--interactive` mode.